core/utils/toolbox.js

```diff
@@ -228,10 +263,56 @@ Blockly.utils.toolbox.toolboxXmlToJson_ = function(toolboxDef) {
  * @package
  */
 Blockly.utils.toolbox.hasCategories = function(toolboxDef) {
+  var toolboxKind = toolboxDef['kind'];
+  if (toolboxKind) {
+    if (toolboxKind != Blockly.utils.toolbox.FLYOUT_TOOLBOX_KIND &&
```

Comment:
There isn't currently another great place for this. hasCategories is used in convertToolboxToJSON and convertToolboxContentsToJSON which is what we are currently using for our validation. 

That being said it might be worth it to have a standalone validation method, but that doesn't exist yet. I'll look into it more, but would prefer to do it in a different PR since I think it will take a decent amount of shuffling around.

Reply:
SGTM, I think a validation method could be useful but can definitely do it later.